### PR TITLE
Remove unwanted screenshot frames check and add default 1 sec wait to avoid css rendering issues

### PR DESCRIPTION
**Pull request by sunilsurana**
Remove unwanted screenshot frames check and add default 1 sec wait to avoid css rendering issues

### CODE REVIEW

File: src/StoryWrightProcessor/PlayWrightExecutor.ts [hunk 1]
```diff
@@ -99,25 +99,13 @@ export class PlayWrightExecutor {
...
     const timeout = Date.now() + 8000; // WHATEVER REASONABLE TIME WE DECIDE
-    let isBuffEqual: boolean;
     let isBusy: boolean;
     do {
-      prevBuf = buf;
-      await this.page.waitForTimeout(300);
-      buf = await this.page.screenshot();
-      isBuffEqual = buf.equals(prevBuf);
+      await this.page.waitForTimeout(1000);
       isBusy = await this.isPageBusy();
```
Comment (p01) [inline src/StoryWrightProcessor/PlayWrightExecutor.ts:106]: Can we lower the waitForTimeout() and add an `await this.page.waitForLoadState("networkIdle");` instead. This is the clear signal that no network calls are in progress including loading web fonts as we inject the first bits of CSS.

Comment (sunilsurana) [reply src/StoryWrightProcessor/PlayWrightExecutor.ts:109]: 1. We already have await this.page.waitForLoadState("networkIdle"); in isPageBusy() method
2. I tried with very low value of 100ms. They problem was not css or font not downloaded but the issue was the css or elements were not completely rendered and there were few pixels where colors were not applied when we took screenshot. As current assumption is 1 sec to paint the screen. I will try with more values and see where we start to see such issue. But for now can we keep this?
3. Once we have computed style dom diffing we can lower this wait significantly in that case if there is no diff in computed dom but diff in png image we can assume that is due to some pixel not rendered.
4. With this change the total execution time was about 30-40% less then that of earlier one where we took screenshots at multiple intervals

Comment (sunilsurana) [reply src/StoryWrightProcessor/PlayWrightExecutor.ts]: Changed code to also make the timeout value from argument

File: src/StoryWrightProcessor/PlayWrightExecutor.ts [hunk 2]
```diff
@@ -99,25 +100,16 @@ export class PlayWrightExecutor {
...
     let isBusy: boolean;
     do {
-      prevBuf = buf;
-      await this.page.waitForTimeout(300);
-      buf = await this.page.screenshot();
-      isBuffEqual = buf.equals(prevBuf);
+      // Add a default wait for 1sec for css rendring, click or hover activities. 
+      // Ideally the test should be authored in such a way that it should wait for element to be visible and then take screenshot but that gets missed out in most test cases.
+      // Also on hover activities where just some background changes its difficult for test author to write such waiting mechanism hence adding default 1 second wait.
+      await this.page.waitForTimeout(this.options.waitTimeScreenshot);
```
Comment (p01) [inline src/StoryWrightProcessor/PlayWrightExecutor.ts:109]: I insist on `await this.page.waitForLoadState("networkidle");` In the "worst" case scenario it waits 500ms after the last network request, and in the best/most-common scenario it is a no-op.

If you want go for a Promise.all to wait both things in parallel:
```typescript
await Promise.all([
  this.page.waitForTimeout(1000),
  this.page.waitForLoadState("networkidle")
]);
```

File: src/StoryWrightProcessor/PlayWrightExecutor.ts (hunk 1 above)
Comment (p01) [reply src/StoryWrightProcessor/PlayWrightExecutor.ts]: I insist on `await this.page.waitForLoadState("networkidle");` In the "worst" case scenario it waits 500ms after the last network request, and in the best/most-common scenario it is a no-op.

If you want go for a Promise.all to wait both things in parallel:
```typescript
await Promise.all([
  this.page.waitForTimeout(1000),
  this.page.waitForLoadState("networkidle")
]);
```

Comment (sunilsurana) [reply src/StoryWrightProcessor/PlayWrightExecutor.ts]: We are already doing that [here](https://github.com/microsoft/storywright/blob/6bdc662cb9aa909d0fb40dc803593ef37253a680/src/StoryWrightProcessor/PlayWrightExecutor.ts#L82)

Also i added couple on log lines on my dev setup and it seems its not waiting for 500ms. I know documentation says that. Am i missing something?

image

Here is the output
Before 1640165861316
After 1640165861317
Before 1640165861320
After 1640165861321
Before 1640165862330
After 1640165862331
Before 1640165862332
After 1640165862333

Comment (sunilsurana) [reply src/StoryWrightProcessor/PlayWrightExecutor.ts]: Not sure if these are related
https://github.com/microsoft/playwright/issues/662#issuecomment-876608105
https://github.com/microsoft/playwright/issues/2515#issuecomment-724163391

**Review by p01 — APPROVED** [review]

**Pull request merged**